stl/inc/future

```diff
@@ -1416,6 +1416,18 @@ private:
     _MyPromiseType _MyPromise;
 };
 
+#if _HAS_CXX17
+#define _FUNCTION_POINTER_DEDUCTION_GUIDE(CALL_OPT, X1, X2, X3) \
+    template <class _Ret, class... _Types>                      \
+    packaged_task(_Ret(CALL_OPT*)(_Types...)) -> packaged_task<_Ret(_Types...)>; // intentionally discards CALL_OPT

```

Comment:
Given that there is only one invocation of that macro I think it is best to simply spell it out?

Reply:
Yes, it's reasonable. Thanks

Reply:
That macro is the argument for another macro, so I can't notice any other elegant way to pass it as parameter. Tell if you have variant, please

Reply:
The `_NON_MEMBER_CALL` machinery is needed to stamp this out for every possible calling convention, which is highly dependent on the platform and compiler options. I know of no practical way to avoid macros here (and as these are the Worst Macros In The Entire STL, I have been very motivated to look for alternative solutions).